flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/TableEnvironmentImpl.java

```diff
@@ -833,10 +832,44 @@ public TableResultInternal executeInternal(List<ModifyOperation> operations) {
         return executeInternal(transformations, sinkIdentifierNames, jobStatusHookList);
     }
 
-    private ModifyOperation getOperation(ReplaceTableAsOperation rtasOperation) {
-        // rtas drop table first, then create
+    private ModifyOperation getModifyOperation(
+            ReplaceTableAsOperation rtasOperation, List<JobStatusHook> jobStatusHookList) {
         CreateTableOperation createTableOperation = rtasOperation.getCreateTableOperation();
         ObjectIdentifier tableIdentifier = createTableOperation.getTableIdentifier();
+        Catalog catalog = catalogManager.getCatalog(tableIdentifier.getCatalogName()).orElse(null);
+        ResolvedCatalogTable catalogTable =
+                catalogManager.resolveCatalogTable(createTableOperation.getCatalogTable());
+        Optional<DynamicTableSink> stagingDynamicTableSink =
+                getSupportsStagingDynamicTableSink(createTableOperation, catalog, catalogTable);
+        if (stagingDynamicTableSink.isPresent()) {
+            // use atomic rtas
+            DynamicTableSink dynamicTableSink = stagingDynamicTableSink.get();
+            if (!rtasOperation.isCreateOrReplace()) {
+                Optional<ContextResolvedTable> driverTable =
```

Comment:
Why name `driverTable`? It really confused me.
Use `oldTable`?

Reply:
change to `replacedTable`